examples/js/CSVnVRTParser.js

```diff
@@ -0,0 +1,207 @@
+                res.attributes[attributes[i]] = xml.getAttributeNode(attributes[i]).value;
+            }
+        }
+
+        if (xml.childElementCount > 0) {
+            for (var i = 0; i < xml.childElementCount; i++) {
+                xml2json(xml.children[i], res);
+            }
+        } else if (xml.textContent) {
+            res.value = xml.textContent;
+        }
+
+        var name = xml.nodeName;
+
+        if (!json[name]) {
+            json[name] = res;
+        } else if (json[name].length > 0) {
+            json[name].push(res);
+        } else {
+            json[name] = [res];
+        }
+
+        return json;
+    }
+
+    function getGeometryType(type) {
+        switch (type) {
+            case 'wkbPoint':
+            case 'wkbMultiPoint':
+                return itowns.FEATURE_TYPES.POINT;
+            case 'wkbLineString':
+            case 'wkbMultiLineString':
+                return itowns.FEATURE_TYPES.LINE;
+            case 'wkbPolygon':
+            case 'wkbMultiPolygon':
+                return itowns.FEATURE_TYPES.POLYGON;
+            default:
+                throw new Error('This type of GeometryType is not supported yet: ' + type);
+                break;
+        }
+    }
+
+    function OGRVRTLayer2Feature(layer, data, crs, options) {
+        var crs = (layer.LayerSRS && layer.LayerSRS.value) || crs;
+
+        var type = itowns.FEATURE_TYPES.POINT;
+        if (layer.GeometryType) {
+            type = getGeometryType(layer.GeometryType.value);
+        }
+
+        var feature = new itowns.Feature(type, options.crsOut, options);
+
+        if (layer.Field) {
+            if (!layer.Field.length) {
+                layer.Field = [layer.Field];
+            }
+
+            for (var f = 0; f < layer.Field.length; f++) {
+                layer.Field[f].attributes.pos = data.header.indexOf(layer.Field[f].attributes.src);
+            }
+        }
+
+        if (layer.GeometryField) {
+            switch (layer.GeometryField.attributes.encoding) {
+                case 'PointFromColumns':
+                    var x = data.header.indexOf(layer.GeometryField.attributes.x);
+                    var y = data.header.indexOf(layer.GeometryField.attributes.y);
+                    var z = data.header.indexOf(layer.GeometryField.attributes.z);
+                    var m = data.header.indexOf(layer.GeometryField.attributes.m);
+
+                    var line;
+                    for (var i = 0; i < data.content.length; i++) {
+                        line = data.content[i];
+                        var geometry = feature.bindNewGeometry();
+
+                        if (layer.Field) {
+                            for (var f = 0; f < layer.Field.length; f++) {
+                                geometry.properties[layer.Field[f].attributes.name] = line[layer.Field[f].attributes.pos];
+                            }
+                        }
+
+                        geometry.startSubGeometry(1);
```

Comment:
It's points cloud?

Reply:
What do you mean ? If I'm not mistaken, in this combination (csv + vrt), there are only points, so I'm adding each in a geometry (or maybe I'm confused with the way `Feature` works ?)